src/Library/demos/Column View/main.vala

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env -S vala workbench.vala --pkg libadwaita-1
+
+// Define our class for our custom model
+public class Book : Object {
+  public string title { get; set; }
+  public string author { get; set; }
+  public int year { get; set; }
+
+  public Book (string title, string author, int year) {
+    Object (
+      title: title,
+      author: author,
+      year: year
+    );
+  }
+}
+
+public void main () {
+  var column_view = (Gtk.ColumnView) workbench.builder.get_object ("column_view");
+  var col1 = (Gtk.ColumnViewColumn) workbench.builder.get_object ("col1");
+  var col2 = (Gtk.ColumnViewColumn) workbench.builder.get_object ("col2");
+  var col3 = (Gtk.ColumnViewColumn) workbench.builder.get_object ("col3");
+
+  // Create the data model
+  var data_model = new ListStore (typeof (Book));
+  data_model.splice (0, 0, {
```

Comment:
Can you put a comment here to explain the usage of `splice()`? I think its not that obvious. Maybe even using `append()` in a foreach loop on an array would be better instead.